finquant/minimise_fun.py

```diff
@@ -3,37 +3,65 @@
 """
 
 
+from finquant.data_types import ARRAY_OR_DATAFRAME, ARRAY_OR_SERIES, FLOAT, NUMERIC
 from finquant.quants import annualised_portfolio_quantities
+from finquant.type_utilities import type_validation
 
 
-def portfolio_volatility(weights, mean_returns, cov_matrix):
+def portfolio_volatility(
+    weights: ARRAY_OR_SERIES[FLOAT],
+    mean_returns: ARRAY_OR_SERIES[FLOAT],
+    cov_matrix: ARRAY_OR_DATAFRAME[FLOAT],
+) -> FLOAT:
     """Calculates the volatility of a portfolio
 
-    :Input:
-     :weights: numpy.ndarray, weights of the stocks in the portfolio
-     :mean_returns: pandas.Series, individual expected returns for all stocks
-         in the portfolio
-     :cov_matrix: pandas.DataFrame, covariance matrix of returns
+    :param weights: An array of weights
+    :type weights: :py:data:`~.finquant.data_types.ARRAY_OR_SERIES`
 
-    Output:
-     :volatility: annualised volatility
+    :param mean_returns: An array of individual expected returns for all stocks
+    :type mean_returns: :py:data:`~.finquant.data_types.ARRAY_OR_SERIES`
+
+    :param cov_matrix: Covariance matrix of returns
+    :type cov_matrix: :py:data:`~.finquant.data_types.ARRAY_OR_DATAFRAME`
+
+    :rtype: :py:data:`~.finquant.data_types.FLOAT`
+    :return: Annualised volatility
     """
+    # Type validations:
+    type_validation(weights=weights, means=mean_returns, cov_matrix=cov_matrix)
```

Comment:
In this line we call the new function `type_validation` that takes care of validating the types of given arguments. Once this PR is merged, please follow this practice as well.